src/Symfony/Component/Finder/Gitignore.php

```diff
@@ -25,12 +25,22 @@ class Gitignore
      * Format specification: https://git-scm.com/docs/gitignore#_pattern_format
      */
     public static function toRegex(string $gitignoreFileContent): string
+    {
+        return self::buildRegex($gitignoreFileContent, false);
+    }
+
+    public static function toRegexMatchingNegatedPatterns(string $gitignoreFileContent): string
```

Comment:
Is this really needed and does it works with nested include/exclude? Is this how standard `git` does the matching?

Reply:
This is needed to support patterns negated in subdirectories, e.g.:
```
# .gitignore
*.txt
```
```
# a/.gitignore
!b.txt
```
We need to match negated patterns separately to know that `a/b.txt` must be included in the result despite being excluded by the rule in the first file. The first regexp would not exclude the file but it would not let us know the file must be included (overriding a previous rule from another file).

This is covered by `VcsIgnoredFilterIterator` tests.

Reply:
I see, very nice!